### PR TITLE
Top n populated cities

### PR DESCRIPTION
**Pull request by LukasP23**
4.1 - 4.5 implemented

### CODE REVIEW

File: src/main/java/com/napier/sem/DAO.java [hunk 1]
```diff
@@ -193,6 +193,47 @@ public DAO(Connection connection)
...
+            case DISTRICT:
+                whereClause = "WHERE city.district = '" + areaName + "'\n";
+                break;
+            default:
+                whereClause = "";
+                break;
+        }
+
+        // Define the SQL query as a string
+        String statementString = "SELECT ID, city.name, district, city.population, country.name AS country\n" +
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java]: we don't need `ID` in the select, as we don't use it. i'm happy to leave it for this though, there's no harm in it.

Comment (LukasP23) [reply src/main/java/com/napier/sem/DAO.java]: Agreed. Removed it in the latest commit.

File: src/main/java/com/napier/sem/DAO.java [hunk 2]
```diff
@@ -193,6 +193,47 @@ public DAO(Connection connection)
...
+                whereClause = "";
+                break;
+        }
+
+        // Define the SQL query as a string
+        String statementString = "SELECT city.name, district, city.population, country.name AS country\n" +
+                "FROM city\n" +
+                "    JOIN country ON city.countrycode = country.code\n" +
+                "WHERE city.population > 0 \n" +
+                whereClause +
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java:231]: lovely

**Review by scan-lan — APPROVED** [review]
psychedelic bro !

**Pull request merged**